core/src/main/java/com/aem/demo/core/services/impl/RssFeedServiceImpl.java

```diff
@@ -41,4 +51,87 @@ public List<NewsCard> readFeed() {
+            Session session = resolver.adaptTo(Session.class);
+
+            if (session.nodeExists(NODE_PATH)){
+                nodeCreated = addNewsCardNode(session, cardList);
+            } else {
+                addParentNode(session);
+                nodeCreated = addNewsCardNode(session, cardList);
+            }
+
+        } catch (Exception e) {
+            log.error("\n Error while creating node - {} ",e.getMessage());
+        }
+        return nodeCreated;
+    }
+
+    private int count = 0;
+
+    private String addNewsCardNode(Session session, List<NewsCard> list){
+
+        try {
+            for (NewsCard elm:list) {
+                count++;
+                String nodeName= count + " newscard_node";
+                Node node = session.getNode(NODE_PATH);
+                Node cardNode = node.addNode(nodeName, NEWSCARDNODE_TYPE);
+                cardNode.setProperty("topic", elm.getTopic());
+                cardNode.setProperty("article", elm.getArticle());
+                cardNode.setProperty("link", elm.getLink());
+                cardNode.setProperty("pubDate", String.valueOf(elm.getPubDate()));
+                cardNode.setProperty("image", elm.getImage());
+                saveImage(cardNode, elm.getImage());
+                session.save();
+            }
+
+
+        } catch (Exception e) {
+            log.error("\n Error while creating NewsCard node ");
+        }
+        return null;
+    }
+
+    private String addParentNode(Session session){
+        try {
+            if(session.nodeExists("/content/aemtraining/us/en")){
+                Node gParentNode=session.getNode("/content/aemtraining/us/en");
+                Node parentNode=gParentNode.addNode("us-newscard",NEWSCARDNODE_TYPE);
+                session.save();
+                return parentNode.getName();
+            }
+        }catch (Exception e){
+            log.error("\n Error while creating Parent node ");
+        }
+        return null;
+    }
+
+    private boolean saveImage(Node node, String link) {
+        try {
+            ResourceResolver resourceResolver = ResolverUtil.newResolver(resourceResolverFactory);
+
+            URL url = new URL(link);
+            URLConnection connection = url.openConnection();
+            InputStream stream = connection.getInputStream();
+            Session session=resourceResolver.adaptTo(Session.class);
+            ValueFactory valueFactory=session.getValueFactory();
+            Binary imageBinary = valueFactory.createBinary(stream);
+            Node photo=node.addNode("photo","sling:Folder");
+            Node file=photo.addNode("image","nt:file");
+            Node content = file.addNode("jcr:content", "nt:resource");
+            content.setProperty("jcr:mimeType", "image/jpeg");
+            content.setProperty("jcr:data", imageBinary);
+
+        } catch (Exception e) {
+            log.info("\n ERROR while saving images - {} ",e.getMessage());
```

Comment:
(cosmetic) "Image", not "images", there's only one image in the context of this method